### PR TITLE
Newsynonyms endpoints

### PR DESCRIPTION
**Pull request by Nagriar**

### CODE REVIEW

**Comment by Nagriar** [comment]
@CBaptiste We can now merge it :)

**Comment by CBaptiste** [comment]
Nope @Nagriar it has conflicts :(

**Comment by Nagriar** [comment]
Arf ok, I'll rebase the branch

**Comment by Nagriar** [comment]
@CBaptiste now it's good 👍

**Comment by CBaptiste** [comment]
waiting for appVeyor and then I ll merge it

**Comment by CBaptiste** [comment]
👍

File: Algolia.Search/Index.cs [hunk 1]
```diff
@@ -1000,5 +1000,199 @@ public JObject SearchDisjunctiveFaceting(Query query, IEnumerable<string> disjun
...
+        public Task<JObject> SaveSynonymAsync(string objectID, object content, bool forwardToSlaves = false, CancellationToken token = default(CancellationToken))
+        {
+            return _client.ExecuteRequest(AlgoliaClient.callType.Write, "PUT", string.Format("/1/indexes/{0}/synonyms/{1}?forwardToSlaves={2}", _urlIndexName,  Uri.EscapeDataString(objectID), forwardToSlaves ? "true" : "false"), content, token);
+        }
+
+        /// <summary>
+        /// Synchronously call <see cref="Index.SaveSynonymAsync"/>.
+        /// </summary>
+        /// <param name="forwardToSlave">Forward the operation to the slave indices</param>
+        public JObject SaveSynonym(string objectID, object content, bool forwardToSlaves = false, bool replaceExistingSynonyms = false)
```
Comment (olance) [inline Algolia.Search/Index.cs]: `replaceExistingSynonyms` is not used at all here, and is not a valid param for the PUT request that is sent for `SaveSynonym`

**Pull request merged**